main/server.go

```diff
@@ -0,0 +1,313 @@
+// marshaled.
+// Importantly, this function only works with files corresponding to JSON
+// objects! Files that correspond to JSON arrays will fail!
+// Also, if more than one file shares the same key, only the value from
+// the final file will be present in the output.
+func mergeJSONOrDie(files ...string) []byte {
+	if len(files) == 0 {
+		panic(fmt.Errorf("Called with no files!"))
+	}
+	b, err := ioutil.ReadFile(files[0])
+	if err != nil {
+		panic(err)
+	}
+	var base map[string]interface{}
+	if err = json.Unmarshal(b, &base); err != nil {
+		panic(err)
+	}
+	for _, file := range files[1:] {
+		b, err = ioutil.ReadFile(file)
+		if err != nil {
+			panic(err)
+		}
+		var next map[string]interface{}
+		if err = json.Unmarshal(b, &next); err != nil {
+			panic(err)
+		}
+		for k, v := range next {
+			base[k] = v
+		}
+	}
+	b, err = json.Marshal(base)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+// errorHandler wraps a function returning an error by handling the error and returning a http.Handler.
+// If the error is of the one of the types defined above, it is handled as described for every type.
+// If the error is of another type, it is considered as an internal error and its message is logged.
+func errorHandler(f func(w http.ResponseWriter, r *http.Request, c context.Context) error) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		c := appengine.NewContext(r)
+		err := f(w, r, c)
+		if err == nil {
+			return
+		}
+		switch err.(type) {
+		case badRequest:
+			http.Error(w, err.Error(), http.StatusBadRequest)
+		case notFound:
+			http.Error(w, "todo not found", http.StatusNotFound)
+		default:
+			log.Errorf(c, "internal error %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	}
+}
+
+// writeJSON marshals v into its JSON representation and writes that to w
+// along with the given status code. It also takes care to set the HTTP
+// Content-Type header. Any error encountered during marshalling is returned.
+func writeJSON(w http.ResponseWriter, statusCode int, v interface{}) error {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(statusCode)
+	return json.NewEncoder(w).Encode(v)
+}
+
+// WriteLearnJSON writes a sidebar JSON file consumable by the TodoMVC
+// JS which is specific to our Go backend implementation.
+func WriteLearnJSON(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.Write(learnJSON)
+}
+
+// IsApiEnabled writes an HTTP 200 indicating that the TodoMvc API is enabled for this app.
+func IsApiEnabled(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+}
+
+// badRequest should be returned by a handler function to indicate a malformed request.
+type badRequest struct{ error }
+
+// notFound should be returned by a handler function to indicate that the requested data could not be found.
+type notFound struct{ error }
```

Comment:
you're never using the error here, so maybe replace with:

var notFound = fmt.Errorf("task not found")